tools/ktx/basis_utils.h

```diff
@@ -317,41 +271,26 @@ struct OptionsCodec {
                 "Default is 18.0. Larger values expand the range of blocks considered smooth.",
                 cxxopts::value<float>(), "<deviation>")
             (kUastcRdoF, "Do not favor simpler UASTC modes in RDO mode.")
-            (kUastcRdoM, "Disable RDO multithreading (slightly higher compression, deterministic).");
-        opts.add_options("Encode common")
-            (kNormalMode, "Optimizes for encoding textures with normal data. If the input texture has "
-                "three or four linear components it is assumed to be a three component linear normal "
-                "map storing unit length normals as (R=X, G=Y, B=Z). A fourth component will be ignored. "
-                "The map will be converted to a two component X+Y normal map stored as (RGB=X, A=Y) "
-                "prior to encoding. If unsure that your normals are unit length, use --normalize. "
-                "If the input has 2 linear components it is assumed to be an X+Y map of unit normals.\n"
-                "The Z component can be recovered programmatically in shader code by using the equations:\n"
-                "    nml.xy = texture(...).ga;              // Load in [0,1]\n"
-                "    nml.xy = nml.xy * 2.0 - 1.0;           // Unpack to [-1,1]\n"
-                "    nml.z = sqrt(1 - dot(nml.xy, nml.xy)); // Compute Z\n"
-                "ETC1S / BasisLZ encoding, RDO is disabled (no selector RDO, no endpoint RDO) to provide better quality.")
-            (kThreads, "Sets the number of threads to use during encoding. By default, encoding "
-                "will use the number of threads reported by thread::hardware_concurrency or 1 if "
-                "value returned is 0.", cxxopts::value<uint32_t>(), "<count>")
+            (kUastcRdoM, "Disable RDO multithreading (slightly higher compression, deterministic).")
             (kNoSse, "Forbid use of the SSE instruction set. Ignored if CPU does "
```

Comment:
I think this is wrong. Although it looks like `kNoSse` is left where it is. It has actually moved from "Encode common" to "Encode UASTC". Is it Also not applicable to "Encode BasisLZ"?

Reply:
@MarkCallow going back to this conversation. I think `--no-sse` should be in the "Encode common options:" because its help text already says it only applies to basis-lz and uastc.

```
--no-sse                  Forbid use of the SSE instruction set. Ignored if CPU does not
                                 support SSE. SSE can only be disabled on the basis-lz and uastc
                                 compressors.
```

We can further add, if not obvious, that its ignore for ASTC.

As it is the other options are. We either create another "Basis encode common options" or we add `--no-sse` to both "Encode UASTC" and "Encode BasisLZ" both doesn't look right.

Reply:
I thought we were going to have an EncodeBasis utils which covered both BasisLZ and UASTC. If you are making separate then it is probably better to put `--no-sse` in EncodeCommon that create a new EncodeBasisCommon. Are there no other options that common to BasisLZ and UASTC but not ASTC?

Reply:
This also means no test/cts changes are required

Reply:
> I thought we were going to have an EncodeBasis utils which covered both BasisLZ and UASTC.

Yes we do. We have renamed `tools/ktx/encode_utils.h` to `tools/ktx/basis_utils.h` and basis_utils.h has all the basis specific options. The problem is it has if we put `no-sse` there it has to either go inside both the sections of "Encode UASTC" and "Encode BasisLZ" or added into a newer "Basis encode common options" as I mentioned above.

If we don't keep `no-sse` inside basis_utils.h then it goes into encode_utils_common.h and we say this is only basis specific options but at least its common between "BasisLZ" and "UASTC".

Reply:
> If we don't keep no-sse inside basis_utils.h then it goes into encode_utils_common.h and we say this is only basis specific options but at least its common between "BasisLZ" and "UASTC".

If  --no-sse` is in EncodeCommonOptions and the ASTC encode command has `Combine(..., EncodeCommonOptions, ...)` the options parser will allow it. We need to ensure an unknown option error is raised if it is specified. This could be either a specific test in the command or a template for EncodeCommonOptions with a boolean set to tell it to allow, or disallow, `--no-sse`.

Reply:
@MarkCallow I assume you mean `OptionsCodecCommon`? I am curious if we really need these checks? The help text already say  its ignored if CPU does not support SSE and can only be disabled on basis-lz or uastc only.

In general also curious what is the use of this flag? why would one want to disable that? is this for testing purposes?

Reply:
Also if you are talking about `ktx encode-astc` tool. I am not clear on that yet, referring to your email about updated doxygen. Are we going to add ASTC encode support to `ktx encode` instead of creating another `ktx encode-astc`?

If thats the case then I suggest we fix this in https://github.com/KhronosGroup/KTX-Software/pull/810

Reply:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/817/commits/84759857c4d7a5a537369157bd84cebd6af3d996

Reply:
In the case of ASTC, even when the CPU supports SSE, `--no-sse` will have no effect. I prefer to have a check to inform the user in order to cut off false bug reports.

I don't know why someone would want to disable SSE. It is one of the BasisU encoder options and we chose to expose all options. Best to ask Rich about it. `basisu` tool's help does not give a reason.

I haven't decided yet about `astc-encode` vs adding ASTC support to `encode`. The updated man page is an experiment to give a feel for what the latter might look like. It doesn't look too bad and is similar to `create` except for having `--codec` instead of `--encode` for BasisU. I am leaning towards this solution but am willing to continue discussion.

Reply:
Please modify the documentation for `--no-sse`, line 57 in `encode_utils_common.h`, adding a sentence "Ignored for other compressors."

Reply:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/817/commits/cf0617ef5e170cc6c095e26e36e02eeed831bbcf although the line number for  me was 55.